### PR TITLE
Fix `selector-pseudo-class-no-unknown` false positives for some `moz-*` vendor-prefixed pseudo-classes

### PR DESCRIPTION
**Pull request by Mouvedia**
> Which issue, if any, is this issue related to?

N/A

> Is there anything in the PR that needs further explanation?

https://bugzilla.mozilla.org/show_bug.cgi?id=309067
https://bugzilla.mozilla.org/show_bug.cgi?id=71261
https://bugzilla.mozilla.org/show_bug.cgi?id=1839221
https://developer.mozilla.org/en-US/docs/Web/CSS/:dir
https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only
https://developer.mozilla.org/en-US/docs/Web/CSS/:read-write

### CODE REVIEW

File: .changeset/thin-buses-greet.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"stylelint": patch
+---
+
+Fixed: `selector-pseudo-class-no-unknown` false positives
```
Comment (Mouvedia) [inline .changeset/thin-buses-greet.md:5]: https://github.com/stylelint/stylelint/blob/main/.changeset/dirty-fishes-approve.md will be removed

File: lib/utils/isAutoprefixable.cjs [hunk 2]
```diff
@@ -237,6 +238,7 @@ const PROPERTY_VALUES = new Set([
 	'-moz-radial-gradient',
 	'-moz-repeating-linear-gradient',
 	'-moz-repeating-radial-gradient',
+	'-moz-text',
```
Comment (Mouvedia) [inline lib/utils/isAutoprefixable.cjs:241]: `user-select` property

File: lib/utils/isAutoprefixable.cjs [hunk 3]
```diff
@@ -221,6 +221,7 @@ const PROPERTIES = new Set([
 const PROPERTY_VALUES = new Set([
 	'-moz-all',
 	'-moz-calc',
+	'-moz-center',
```
Comment (Mouvedia) [inline lib/utils/isAutoprefixable.cjs:224]: `text-align` property

File: lib/utils/isAutoprefixable.mjs [hunk 4]
```diff
@@ -233,6 +234,7 @@ const PROPERTY_VALUES = new Set([
 	'-moz-radial-gradient',
 	'-moz-repeating-linear-gradient',
 	'-moz-repeating-radial-gradient',
+	'-moz-text',
```
Comment (Mouvedia) [inline lib/utils/isAutoprefixable.mjs:237]: for `user-select` property

File: lib/reference/selectors.mjs [hunk 5]
```diff
@@ -235,21 +235,28 @@ const vendorSpecificPseudoClasses = new Set([
...
 	'-moz-meter-optimum',
 	'-moz-meter-sub-optimum',
 	'-moz-meter-sub-sub-optimum',
 	'-moz-only-whitespace',
 	'-moz-placeholder',
+	'-moz-read-only',
+	'-moz-read-write',
 	'-moz-submit-invalid',
 	'-moz-suppressed',
+	'-moz-table-border-nonzero',
```
Comment (Mouvedia) [inline lib/reference/selectors.mjs:259]: `[border]:not([border="0"])`

**Comment by ybiquitous** [comment]
@Mouvedia I'd appreciate making bugfix PRs. But I was hoping you could create a dedicated PR so we can easily review a PR and track the commit history unless there is no special reason. I believe it's not a good sign that one PR includes multiple changelog items.

For an example of [`value-no-vendor-prefix`](https://stylelint.io/user-guide/rules/value-no-vendor-prefix), we could see the [Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLjmAHjgLRIA2AlgOabwrEC2EAXsWDJvgE4DcWu+IqUo06xAO4wARgGsKJdpxi8sAXywgANCABmFMjABySBnESETABwMA6MAGd7W8BEx6qCEBkwoU6EIT4mFD2-nQA2li+vv72OACeBpScbK7uxHFIwUhcUP5RKAC6as6QbtQAYhBcDEg4ngBW9q7OsJZOiN4xIHGJMMk4YX4gZHUwcf6aBbEJSRQpZemZ2blD-qP4EyAlqkA) to reproduce the bug and helpful resources, such as MDN links.

File: .changeset/dirty-fishes-approve.md
Comment (Mouvedia) [inline .changeset/dirty-fishes-approve.md:1]: has the same author
i.e. me

File: .changeset/thin-buses-greet.md (hunk 1 above)
Comment (jeddy3) [inline .changeset/thin-buses-greet.md:5]:
```suggestion
Fixed: `selector-pseudo-class-no-unknown` false positives for some `moz-*` vendor-prefixed pseudo-classes
```

**Review by jeddy3 — COMMENTED** [review]
One minor suggestion to add some details to the changelog entry, otherwise LGTM thank you.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

**Pull request merged**